src/frontend/src/app/scheduled-process/scheduled-process.component.ts

```diff
@@ -7,48 +7,134 @@ import { SchedulesService } from '../services/index';
   }
 
-  public getSchedules():void {
-    this.schedulesData = [];
+  public getSchedules(): void {
+    this.scheduleData = [];
     this.schedulesService.getSchedule().
       subscribe(
       data => {
-        this.schedulesData = data.schedules;
-        console.log("This is the getSchedules ",  data.schedules);
+        this.scheduleData = data.schedules
+        console.log("This is the getSchedule ", data.schedules)
       },
       error => { console.log("error", error) });
   }
 
-  public getSchedulesProcesses():void {
-    this.schedulesProcessData = [];
+  public getSchedulesProcesses(): void {
+    this.scheduleProcess = [];
     this.schedulesService.getScheduledProcess().
       subscribe(
       data => {
-        this.schedulesProcessData = data.processes;
-        console.log("This is the getSchedulesProcesses ",  this.schedulesProcessData);
+        this.scheduleProcess = data.processes
+        console.log("This is the getScheduleProcess ", this.scheduleProcess)
       },
       error => { console.log("error", error) });
   }
 
-  public getLatestTasks():void {
+  public getLatestTasks(): void {
     this.tasksData = [];
     this.schedulesService.getLatestTask().
       subscribe(
       data => {
         this.tasksData = data.tasks;
-        console.log("This is the tasks ",  data.tasks);
+        console.log("Tasks data", data.tasks)
       },
-      error => { console.log("error", error) });
+      error => { console.log("error", error) })
+  }
+
+  /**
+   *  To set schedule type key globally for required field handling on UI
+   * @param value  
+   */
+  public setScheduleTypeKey(value) {
+    this.selected_schedule_type = value;
+  }
+
+  public createSchedule() {
+    let schedule_name = <HTMLInputElement>document.getElementById("name")
+    let schedule_process = <HTMLSelectElement>document.getElementById("process")
+    let schedule_type = <HTMLSelectElement>document.getElementById("type")
+    let repeat_interval = <HTMLInputElement>document.getElementById("repeat")
+    let exclusive_state = <HTMLInputElement>document.getElementById("exclusive")
+
+    // "schedule_type": [{"index": 1, "name": "STARTUP"},{"index": 2,"name": "TIMED"},
+    // {"index": 3,"name": "INTERVAL"},{"index": 4,"name": "MANUAL"}]
+    // For schedule type 'TIMED', show 'Day' and 'TIME' field on UI
+    if (this.selected_schedule_type == 2) {
+      var scheduler_day_field = <HTMLSelectElement>document.getElementById("day")
+      var scheduler_time_field = <HTMLInputElement>document.getElementById("time")
+      var day = scheduler_day_field.value
+      var time = scheduler_time_field.value
+      var scheduled_time = time != '' ? this.converTimeToSec(time) : 0
+    }
+    var repeat_time = repeat_interval.value != '' ? this.converTimeToSec(repeat_interval.value) : 0
+    
+    // check if time is in valid range
+    this.invalidRepeat = this.not_between(repeat_time)
+    this.invalidTime = this.not_between(scheduled_time)
```

Comment:
scheduled_time is in `if (this.selected_schedule_type == 2) {` is its not selected then undefined ?!!